source/to_cpp1.h

```diff
@@ -272,6 +272,9 @@ class positional_printer
     )
         -> void
     {
+if (s.starts_with("..")) {
+    s == s;
```

Comment:
This is causing "unused" warnings which lead to errors in the tests on some platforms. Is the `if` statement actually necessary?

Reply:
I've just checked that this is the reason for all failures among the multi-platform build tests.
The failures in the regression tests seem to be mainly due to necessary line updates in error messages. This should be quick to fix, but first the `s == s` needs to be fixed as this might further change line numbers.

Reply:
Ah, I should remove that... that was so I could set a debug breakpoint on 276 that would fire only when `..` was happening, to investigate why `..` wasn't yet lowering to `.` correctly.

Reply:
OK, so let me get the tests running. I will create a pull request into this branch.
Please give me a couple of minutes.

Reply:
Sorry, it took a bit longer than a couple of minutes, but here is the PR into this one: #1103.

Reply:
Thanks! This PR shouldn't affect any regression test output I think? The test generation should be identical, except only the one test I updated to add a `..` test case which has expanded `.cpp` output but should not have any output changes because it doesn't print anything.

Re line# output: Usually the only kind of change that affects regression test line# output is when `cpp2util.h` changes cause the assertion line numbers to change for the handful of tests that a given older compiler doesn't fully support (typically GCC 10, sometimes Clang 12). So I don't think there should be line# regression test impact for this PR... I hope it's a pure extension!

Reply:
Ah, true, you are right.

Reply:
It seems some of the issues solved in #1103 are pre-existent (except the update of `regression-tests/test-results/pure2-ufcs-member-access-and-chaining.cpp`).

Reply:
Done. The tests on #1103 will be failing until you merge it into this branch. You can also merge this one with failures on `pure2-ufcs-member-access-and-chaining.cpp` and then #1103 to fix those.